### PR TITLE
[Static reference] LPS-182671 Remove static references in class MBDiscussionPermission | part 1

### PR DESCRIPTION
**Pull request by W2jx**
Hi @yuhai , I have resent it.

Previous: https://github.com/liferay-core-infra/liferay-portal/pull/4944

### CODE REVIEW

**Comment by W2jx** [comment]
ci:test:lima

**Review by yuhai — APPROVED** [review]
@dantewang, the pr is good to me.

The current pr mainly makes MBDiscussionPermissionImpl as DiscussionPermission service . This is for the next plan "Removing MBDiscussionPermission class". We will create one new MBDiscussionModelResourcePermission as ModelResourcePermission service. In this MBDiscussionModelResourcePermission, it will reference DiscussionPermission to check permission.

**Comment by Kvale1733** [comment]
ci:test:comment

**Comment by Kvale1733** [comment]
ci:test:message-boards

File: portal-kernel/src/com/liferay/portal/kernel/comment/BaseDiscussionPermission.java [hunk 1]
```diff
@@ -35,7 +35,8 @@ public void checkAddPermission(
 				permissionChecker, companyId, groupId, className, classPK)) {
 
 			throw new PrincipalException.MustHavePermission(
-				0, className, classPK, ActionKeys.ADD_DISCUSSION);
+				permissionChecker, className, classPK,
```
Comment (dantewang) [inline portal-kernel/src/com/liferay/portal/kernel/comment/BaseDiscussionPermission.java:38]: @W2jx This commit says "use correct user id" but it changes "0" to `permissionChecker`. Could you update the message to be more clear?

File: modules/apps/message-boards/message-boards-comment/src/main/java/com/liferay/message/boards/comment/internal/MBCommentManagerImpl.java [hunk 2]
```diff
@@ -319,10 +318,8 @@ public Discussion getDiscussion(
 	}
 
 	@Override
-	public DiscussionPermission getDiscussionPermission(
-		PermissionChecker permissionChecker) {
-
-		return new MBDiscussionPermissionImpl(permissionChecker);
+	public DiscussionPermission getDiscussionPermission() {
+		return new MBDiscussionPermissionImpl();
```
Comment (dantewang) [inline modules/apps/message-boards/message-boards-comment/src/main/java/com/liferay/message/boards/comment/internal/MBCommentManagerImpl.java]: @W2jx @yuhai I wonder if this is really necessary. You first change it to a class without constructor, and then change it into a component. Why not directly make it a component and apply?

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by W2jx** [comment]
Hi @dantewang , I have updatet the commit message, and reordered commits.

**Comment by W2jx** [comment]
no logic changes, no need to rerun test

**Review by dantewang — APPROVED** [review]

**Comment by W2jx** [comment]
no conflicts(7/11 13:11)

**Comment by W2jx** [comment]
no conflicts(7/12 8:01)

**Comment by W2jx** [comment]
no conflicts(7/13 15:32)

**Comment by tinatian** [comment]
ci:forward:force

**Comment by W2jx** [comment]
no conflicts(7/17 8:16)

**Comment by W2jx** [comment]
ci:test:help

**Comment by W2jx** [comment]
ci:help

**Comment by W2jx** [comment]
ci:forward:force

**Comment by dantewang** [comment]
ci:forward:force

**Pull request closed without merging**